tests/cross/src/test/scala/tests/pc/InsertInferredTypeSuite.scala

```diff
@@ -267,56 +295,47 @@ class InsertInferredTypeSuite extends BaseCodeActionSuite {
        |""".stripMargin
   )
 
+  val additionalSpace: String = if (isScala3Version(scalaVersion)) " " else ""
   checkEdit(
     "higher-kinded-types",
     """|package io
        |
-       |import cats.Parallel
-       |import cats.effect.ConcurrentEffect
-       |import cats.effect.ContextShift
-       |import cats.effect.IOApp
        |import cats.effect.Resource
-       |import cats.effect.Timer
        |
        |object Main2 extends IOApp {
        |
        |  trait Logger[T[_]]
        |
-       |  def mkLogger[F[_]: ConcurrentEffect: Timer: ContextShift]: Resource[F, Logger[F]] = ???
+       |  def mkLogger[F[_]]: Resource[F, Logger[F]] = ???
        |
-       |  def <<serve>>[F[_]: ConcurrentEffect: ContextShift: Timer: Parallel]() =
+       |  def <<serve>>[F[_]]() =
        |    for {
        |      logger <- mkLogger[F]
        |    } yield ()
        |
        |}
        |""".stripMargin,
-    """|package io
-       |
-       |import cats.Parallel
-       |import cats.effect.ConcurrentEffect
-       |import cats.effect.ContextShift
-       |import cats.effect.IOApp
-       |import cats.effect.Resource
-       |import cats.effect.Timer
-       |
-       |object Main2 extends IOApp {
-       |
-       |  trait Logger[T[_]]
-       |
-       |  def mkLogger[F[_]: ConcurrentEffect: Timer: ContextShift]: Resource[F, Logger[F]] = ???
-       |
-       |  def serve[F[_]: ConcurrentEffect: ContextShift: Timer: Parallel](): Resource[F,Unit] =
-       |    for {
-       |      logger <- mkLogger[F]
-       |    } yield ()
-       |
-       |}
-       |""".stripMargin
+    s"""|package io
+        |
+        |import cats.effect.Resource
+        |
+        |object Main2 extends IOApp {
+        |
+        |  trait Logger[T[_]]
+        |
+        |  def mkLogger[F[_]]: Resource[F, Logger[F]] = ???
+        |
+        |  def serve[F[_]](): Resource[F,${additionalSpace}Unit] =
+        |    for {
+        |      logger <- mkLogger[F]
+        |    } yield ()
+        |
+        |}
+        |""".stripMargin
   )
 
   checkEdit(
-    "path",
+    "path".tag(IgnoreScala3),
```

Comment:
This is the last problematic case, it's depended on fixing the `shortType` method in case of conflicts, which doesn't work perfectly currently.